lib/Backend/arm64/LowerMD.cpp

```diff
@@ -1392,47 +1416,75 @@ LowererMD::LowerEntryInstr(IR::EntryInstr * entryInstr)
             IR::RegOpnd::New(nullptr, RegSP, TyMachReg, this->m_func),
             IR::IntConstOpnd::New(0, TyMachReg, this->m_func), this->m_func);
         insertInstr->InsertBefore(instrMov);
-
-        regPushedSize = 0; // STP_PRE Advanced the SP
     }
 
     if (!usedRegs.IsEmpty())
     {
+        IR::RegOpnd* src1 = nullptr;
+        IR::RegOpnd* src2 = nullptr;
         for (RegNum reg = LAST_CALLEE_SAVED_GP_REG; reg > LAST_INT_ARG_REG; reg = (RegNum)(reg - 1))
         {
             if (usedRegs.Test(RegEncode[reg]))
             {
-                regPushedSize += MachRegInt;
-                IR::Instr * instrStore = IR::Instr::New(Js::OpCode::STR, this->m_func);
-                instrStore->SetDst(IR::IndirOpnd::New(IR::RegOpnd::New(nullptr, RegSP, TyMachReg, this->m_func), -regPushedSize, TyMachReg, this->m_func));
-                instrStore->SetSrc1(IR::RegOpnd::New(reg, TyMachReg, this->m_func));
-                insertInstr->InsertBefore(instrStore);
+                if (src1 == nullptr)
+                {
+                    src1 = IR::RegOpnd::New(reg, TyMachReg, this->m_func);
+                }
+                else
+                {
+                    Assert(src2 == nullptr);
+                    src2 = IR::RegOpnd::New(reg, TyMachReg, this->m_func);
+                }
+            }
+
+            if (src1 != nullptr && src2 != nullptr)
+            {
+                IR::Instr * instrStp = IR::Instr::New(Js::OpCode::STP_PRE,
+                    IR::IndirOpnd::New(IR::RegOpnd::New(nullptr, RegSP, TyMachReg, this->m_func), -(2 * MachRegInt), TyMachReg, this->m_func),
+                    src2, src1, this->m_func);
+                insertInstr->InsertBefore(instrStp);
+                src1 = src2 = nullptr;
```

Comment:
Sorry, this won't work with the unwind codes. You must pre-allocate all the space up front, then use STP (no _PRE) to store the values with a positive offset.